api/metric/sdkhelpers.go

```diff
@@ -51,42 +92,38 @@ type BoundInstrumentImpl interface {
 	Unbind()
 }
 
-// WrapInt64CounterInstrument wraps the instrument in the type-safe
-// wrapper as an integral counter.
-//
-// It is mostly intended for SDKs.
-func WrapInt64CounterInstrument(instrument InstrumentImpl, err error) (Int64Counter, error) {
-	common, err := newCommonMetric(instrument, err)
-	return Int64Counter{commonMetric: common}, err
+// AsynchronousImpl is an implementation-level interface to an
+// asynchronous instrument.
+type AsynchronousImpl interface {
+	InstrumentImpl
+
+	// Note: An `Unregister()` API could be supported here.
 }
 
-// WrapFloat64CounterInstrument wraps the instrument in the type-safe
-// wrapper as an floating point counter.
-//
-// It is mostly intended for SDKs.
-func WrapFloat64CounterInstrument(instrument InstrumentImpl, err error) (Float64Counter, error) {
-	common, err := newCommonMetric(instrument, err)
-	return Float64Counter{commonMetric: common}, err
+// wrappedMeterImpl implements the `Meter` interface given a
+// `MeterImpl` implementation.
+type wrappedMeterImpl struct {
+	impl MeterImpl
 }
 
-// WrapInt64MeasureInstrument wraps the instrument in the type-safe
-// wrapper as an integral measure.
-//
-// It is mostly intended for SDKs.
-func WrapInt64MeasureInstrument(instrument InstrumentImpl, err error) (Int64Measure, error) {
-	common, err := newCommonMetric(instrument, err)
-	return Int64Measure{commonMetric: common}, err
+// int64ObserverResult is an adapter for int64-valued asynchronous
+// callbacks.
+type int64ObserverResult struct {
+	observe func(core.Number, LabelSet)
 }
 
-// WrapFloat64MeasureInstrument wraps the instrument in the type-safe
-// wrapper as an floating point measure.
-//
-// It is mostly intended for SDKs.
-func WrapFloat64MeasureInstrument(instrument InstrumentImpl, err error) (Float64Measure, error) {
-	common, err := newCommonMetric(instrument, err)
-	return Float64Measure{commonMetric: common}, err
+// float64ObserverResult is an adapter for float64-valued asynchronous
+// callbacks.
+type float64ObserverResult struct {
+	observe func(core.Number, LabelSet)
 }
 
+var (
+	_ Meter                 = (*wrappedMeterImpl)(nil)
+	_ Int64ObserverResult   = (*int64ObserverResult)(nil)
```

Comment:
Does it need to be a pointer?

Reply:
I removed the use a pointer here. It doesn't need to be one way or another, but there is a small allocation as a result. (I believe a function pointer is greater than a single word, so the aforementioned single-word-object optimization would not have applied.)